src/r1cs/prover.rs

```diff
@@ -448,26 +490,85 @@ impl<'g, T: BorrowMut<Transcript>> Prover<'g, T> {
             return Err(R1CSError::InvalidGeneratorsLength);
         }
 
+        // The shared length can not be longer than the number of inputs itself.
+        if shared_length > 2 * n1 {
```

Comment:
Why * 2?

Reply:
Because n1 is the length of a_L (and a_R) but 2 * n1 is the length of the interleaved a_L and a_R.

Reply:
👍